src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,166 @@
+        return db.collection(COLLECTION_PATH)
+                .document(booking.getId())
+                .set(booking)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String deleteBooking(final String bookingId) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .delete()
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String updateBooking(String bookingId, String endDate, PaymentStatus newStatus) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .update("endDate", endDate, "paymentStatus", newStatus)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    public Scooter bookAScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+        Random random = new Random();
+        Location userLocation = new Location(lat, longitude);
+        Set<Scooter> allAvailableScooters = scooterService.getAvailableScooters(userLocation, 10.0);
+        if (allAvailableScooters.size() > 0) {
+            int randomIndexScooter = random.nextInt(allAvailableScooters.size());
+            Scooter[] listOfScooters = allAvailableScooters.toArray(new Scooter[allAvailableScooters.size()]);
+            Scooter pickedScooter = listOfScooters[randomIndexScooter];
+            pickedScooter.setStatus(ScooterStatus.RESERVED);
+            scooterService.updateScooter(pickedScooter.getSerialNumber(), pickedScooter.getCurrentLocation(), pickedScooter.getStatus(), pickedScooter.getBattery().getLevel());
+            LOGGER.info("scooter:" + pickedScooter.toString());
+            LinkedHashSet<Booking> bookings = getAllBookings();
+            int randomIndexBooking = random.nextInt(bookings.size());
+            Booking[] listOfBookings = bookings.toArray(new Booking[bookings.size()]);
+            Booking pickedBooking = listOfBookings[randomIndexBooking];
+            pickedBooking.setSerialNumber(pickedScooter.getSerialNumber());
+            String startDateBooking = pickedBooking.getStartDate();
+            pickedBooking.setStartDate(startDateBooking);
+            LOGGER.info("booking:" + pickedBooking.toString());
+            createBooking(pickedBooking);
+
+            return pickedScooter;
+        }
+        return null;
+    }
+
+    public Double endBookingAndUpdate(Booking booking, String endDate) throws ExecutionException, InterruptedException {

```

Comment:
Is the return value used? You could return a boolean to signify when there was a payment and updates were performed successfully.